python/dataproc_templates/gcs/README.md

```diff
@@ -266,6 +266,83 @@ export JARS=<gcs-bucket-location-containing-jar-file>
+
+## Arguments
+
+* `gcs.mongo.input.format`: Input file format (one of: avro,parquet,csv,json)
+* `gcs.mongo.input.location`: GCS location of the input files (format: `gs://BUCKET/...`)
+* `gcs.mongo.output.uri`: MongoDB Output URI for connection
+* `gcs.mongo.output.database`: MongoDB Output Database Name
+* `gcs.mongo.output.collection`: MongoDB Output Collection Name
+* `gcs.mongo.output.mode`: Output write mode (one of: append,overwrite,ignore,errorifexists)(Defaults to append)
+* `gcs.mongo.batch.size`: Output Batch Size (Defaults to 512)
+
+## Usage
+
+```
+$ python main.py --template GCSTOMONGO --help
+
+usage: main.py --template GCSTOMONGO [-h] \
+    --gcs.mongo.input.location GCS.MONGO.INPUT.LOCATION \
+    --gcs.mongo.input.format {avro,parquet,csv,json} \
+    --gcs.mongo.output.uri GCS.MONGO.OUTPUT.URI \
+    --gcs.mongo.output.database GCS.MONGO.OUTPUT.DATABASE \        
+    --gcs.mongo.output.collection GCS.MONGO.OUTPUT.COLLECTION \
+    --gcs.mongo.batch.size GCS.MONGO.BATCH.SIZE \
+    --gcs.mongo.output.mode {overwrite,append,ignore,errorifexists}
+
+
+optional arguments:
+  -h, --help            show this help message and exit
+  --gcs.mongo.input.location GCS.MONGO.INPUT.LOCATION
+                        GCS location of the input files
+  --gcs.mongo.input.format {avro,parquet,csv,json}
+                        Input file format (one of: avro,parquet,csv,json)                        
+  --gcs.mongo.output.uri GCS.MONGO.OUTPUT.URI
+                        Driver URI to connect with MongoDB Instance consisting of username and passwprd as well
+  --gcs.mongo.output.database GCS.MONGO.OUTPUT.DATABASE
+                        MongoDB Database Name
+  --gcs.mongo.output.collection GCS.MONGO.OUTPUT.COLLECTION
+                        MongoDB output Collection name
+  --gcs.mongo.batch.size GCS.MONGO.BATCH.SIZE
+                        MongoDB output Batch size
+  --gcs.mongo.output.mode {overwrite,append,ignore,errorifexists}
+                        Output write mode (one of: append,overwrite,ignore,errorifexists) (Defaults to append)                        
+```
+
+## Required JAR files
+
+This template requires the MongoDB-Java Driver jar file to be available in the Dataproc cluster. Aprt from that, MongoDB-Spark connector jar file is also required to Export and Import Dataframe via Spark.
+User has to download both the required jar files and host it inside a GCS Bucket, so that it could be referred during the execution of code.
+Once the jar file gets downloaded, please upload the file into a GCS Bucket.
+
+Wget Command to download these jar files is as follows :-
+
+```
+sudo wget https://repo1.maven.org/maven2/org/mongodb/spark/mongo-spark-connector_2.12/2.4.0/mongo-spark-connector_2.12-2.4.0.jar
+sudo wget https://repo1.maven.org/maven2/org/mongodb/mongo-java-driver/3.9.1/mongo-java-driver-3.9.1.jar
+```
+
+## Example submission
+
+```
+export JARS=<gcs-bucket-location-containing-jar-file> 
```

Comment:
can we add other export commands in the readme?

Reply:
Added other export commands as well.